src/Windows/Avalonia.Win32/ScreenImpl.cs

```diff
@@ -28,9 +29,28 @@ public IReadOnlyList<Screen> AllScreens
                         (IntPtr monitor, IntPtr hdcMonitor, ref Rect lprcMonitor, IntPtr data) =>
                         {
                             MONITORINFO monitorInfo = MONITORINFO.Create();
-                            if (GetMonitorInfo(monitor,ref monitorInfo))
+                            if (GetMonitorInfo(monitor, ref monitorInfo))
                             {
-                                GetDpiForMonitor(monitor, MONITOR_DPI_TYPE.MDT_EFFECTIVE_DPI, out var x, out _);
+                                var dpi = 1.0;
+
+                                var shcore = LoadLibrary("shcore.dll");
+                                var method = GetProcAddress(shcore, nameof(GetDpiForMonitor));
+                                if (method != IntPtr.Zero)
```

Comment:
I think we can cache the result of this check in a static variable. Other than that looks good